starter-code/lib/main.js

```diff
@@ -0,0 +1,95 @@
+// --------- ITERATION 1: EMPLOYEE SATISFACTION ---------- //
+
+// 1) Declare array of possible satisfaction opinions
+var opinions = [ "This is the best job ever",
+                 "Satisfied",
+                 "At least I get paid",
+                 "I'm looking for another job",
+                 "I don't want to answer"];
+
+// 2) Function that randomly selects an item in an array
+function randomSelector(array) {
+  return array[_.random(0,array.length-1)];
+}
+
+// 4) Function that adds a department (opinions) to employee survey
+function addDepartment(array) {
+  return array.push(_.times(10, randomSelector.bind(null, opinions)));
+}
+
+// General employee satisfaction method
+var employeeSatisfaction = function () {
+  var array = [];
+  _.times(5,addDepartment.bind(null, array));
+  return array;
+};
+
+//Display survey;
+console.log(employeeSatisfaction());
+
+
+// --------- ITERATION 2: HARRY POTTER ---------- //
+
+// 1) Bundle names and birthdays
+birthdays = _.chunk(birthdays,2);
+
+
+// 2-3) Mission birthdays to be added + function
+var moreBirthdays = ["Lily Evans", "30 January", "James Potter", "27 March",
+                     "Dudley Dursley", "30 June", "Tom Riddle", "31 December"];
+
+var addBirthdays = function(bdays) {
+  return _.concat(birthdays, _.chunk(bdays,2));
+}
+
+birthdays = addBirthdays(moreBirthdays);
+//console.log(birthdays);
+
+
+// --------- ITERATION 2: THE PASSWORD PROBLEM ---------- //
+
+// Function to make sure no repeat characters
+var noRepeatChar = function(password) {
+  return _.isEqual(_.uniq(password), password.split("")); // Compares uniquified password to itself
+}
+
+// Function to make sure only numbers
+var onlyNumbers = function(password) {
+  var pwd = _.map(password.split(""), _.toNumber); // Convert password string to array of numbers, where not a number => NaN
+  return _.indexOf(pwd, NaN) === -1 ? true:false; // Returns true if no NaN found
```

Comment:
you can also use `filter` to remove all numbers and check if the size is bigger than 0 at the end